### PR TITLE
Strip mirrord's temp dir from file ops operations, read locally

### PR DESCRIPTION
**Pull request by t4lz**
#1403

If the user's application tries to read files from our temp directory, strip our directory from the path, and read the stripped path locally.

### CODE REVIEW

File: mirrord/layer/src/common.rs [hunk 1]
```diff
@@ -16,6 +18,18 @@ use crate::{
     HOOK_SENDER,
 };
 
+lazy_static! {
```
Comment (aviramha) [inline mirrord/layer/src/common.rs:21]: put it in `sip` crate so we'll also have it only on macOS

**Review by aviramha — COMMENTED** [review]
Overall comment - I think the implementation can be that you just edit the original string, as it will always be less characters. It will make the implementation easier.

**Comment by t4lz** [comment]
> just edit the original string

What do you mean? This is already how it's done.

**Comment by aviramha** [comment]
> > just edit the original string
> 
> What do you mean? This is already how it's done.

Aren't you creating a new string then replacing the ptr?

**Comment by aviramha** [comment]
> > just edit the original string
> 
> What do you mean? This is already how it's done.

Oops. Missed that

**Comment by aviramha** [comment]
You should check if the path is of the binary itself, then we should leave it as is.
I wouldn't add a bypass for it, but just change it as part of checked into without changing the behavior - maybe do a bypass if it's that path though

**Comment by aviramha** [comment]
I agree we should bypass on this case, just not a new case of bypass

**Comment by t4lz** [comment]
> I agree we should bypass on this case, just not a new case of bypass

Oh, I see. But I use this bypass variant to pass back the new pointer. How should I do it then?

**Comment by aviramha** [comment]
> > I agree we should bypass on this case, just not a new case of bypass
> 
> Oh, I see. But I use this bypass variant to pass back the new pointer. How should I do it then?

In the checked into?

**Comment by t4lz** [comment]
> In the checked into?

I am returning a `Bypass` with a the new `*const c_char` from `checked_into` if original path is in our dir. How should I do that without introducing a new variant of `Bypass`?

**Comment by aviramha** [comment]
> > In the checked into?
> 
> I am returning a `Bypass` with a the new `*const c_char` from `checked_into` if the original path is in our dir. How should I do that without introducing a new variant of `Bypass`?

I thought `checked_into` requires `self` and not `&self` (because Into does, my bad).

File: mirrord/layer/src/common.rs [hunk 2]
```diff
@@ -16,6 +20,12 @@ use crate::{
     HOOK_SENDER,
 };
 
+#[cfg(target_os = "macos")]
+lazy_static! {
+    /// Path of current executable, None if fetching failed.
+    pub static ref CURRENT_EXE: Option<String> = std::env::current_exe().ok().map(|path_buf| path_buf.to_string_lossy().to_string());
+}
```
Comment (infiniteregrets) [inline mirrord/layer/src/common.rs:27]: Can we not use https://docs.rs/once_cell/latest/once_cell/sync/struct.Lazy.html here

File: mirrord/auth/Cargo.toml [hunk 3]
```diff
@@ -20,7 +20,7 @@ default = []
 webbrowser = ["dep:rand", "dep:reqwest", "dep:webbrowser"]
 
 [dependencies]
-lazy_static = "1"
+lazy_static.workspace = true
 rand = { version = "0.8", optional = true }
```
Comment (infiniteregrets) [inline mirrord/auth/Cargo.toml:24]: same

File: Cargo.toml [hunk 4]
```diff
@@ -51,6 +51,7 @@ num-traits = "0.2"
 regex = "1"
 miette = "5"
 fancy-regex = { version = "0.10" }
+lazy_static = "1"
```
Comment (infiniteregrets) [inline Cargo.toml:55]: no need of these

**Review by infiniteregrets — CHANGES_REQUESTED** [review]

**Pull request merged**